src/chainstate/burn/db/burndb.rs

```diff
@@ -843,674 +648,811 @@ impl BurnDB {
+
+    pub fn evaluate_sortition(&mut self, burn_header: &BurnchainBlockHeader, ops: Vec<BlockstackOperationType>,
+                              burnchain: &Burnchain, pox_id: &PoxIdentifier, pox_db: &PoxDB) -> Result<(BlockSnapshot, BurnchainStateTransition), BurnchainError> {
+        let parent_pox = pox_db.get_canonical_pox_id(&burn_header.parent_block_hash)
+            .map_err(|_e| BurnchainError::MissingParentBlock)?;
+        let parent_sort_id = SortitionId::new(&burn_header.parent_block_hash, &parent_pox);
+
+        if !pox_db.is_pox_id_a_child(&parent_pox, pox_id) {
+            return Err(BurnchainError::NonCanonicalPoxId(parent_pox, pox_id.clone()));
+        }
+
+        let mut sortition_db_handle = SortitionHandleTx::begin(self, &parent_sort_id)?;
+        let parent_snapshot = sortition_db_handle.as_conn().get_block_snapshot(&burn_header.parent_block_hash)?
+            .ok_or_else(|| {
+                warn!("Unknown block {:?}", burn_header.parent_block_hash);
+                BurnchainError::MissingParentBlock
+            })?;
+
+        let new_snapshot = sortition_db_handle.process_block_txs(
+            &parent_snapshot, burn_header, burnchain, ops)?;
+
+        // commit everything!
+        sortition_db_handle.commit()?;
+        Ok(new_snapshot)
+    }
+
+    pub fn is_stacks_block_in_sortition_set(&self, sortition_id: &SortitionId, block_to_check: &BlockHeaderHash) -> Result<bool, BurnchainError> {
+        self.index_handle(sortition_id)
+            .expects_stacks_block_in_fork(block_to_check)
+            .map_err(|e| BurnchainError::from(e))
+    }
+
+    pub fn latest_stacks_blocks_processed(&self, sortition_id: &SortitionId) -> Result<u64, BurnchainError> {
+        let db_handle = self.index_handle(sortition_id);
+        SortitionDB::get_max_arrival_index(&db_handle)
+            .map_err(|e| BurnchainError::from(e))
+    }
+}
+
+// Querying methods
+impl SortitionDB {
+    /// Get the last snapshot processed, in the provided PoX fork
+    pub fn get_last_snapshot(conn: &Connection, pox_id: &PoxIdentifier, pox_db: &PoxDB) -> Result<Option<BlockSnapshot>, db_error> {
+        let qry = "SELECT * FROM snapshots WHERE pox_id = ?1 ORDER BY block_height DESC, burn_header_hash ASC LIMIT 1";
+        let opt_result = query_row(conn, qry, &[pox_id])?;
+        if let None = opt_result {
+            let parent_pox_id = match pox_db.get_parent_pox_id(pox_id).ok() {
+                None => return Ok(None),
+                Some(x) => x
+            };
+            query_row(conn, qry, &[&parent_pox_id])
+        } else {
+            Ok(opt_result)
+        }
+    }
+
+    /// Get the canonical burn chain tip -- the tip of the longest burn chain we know about.
+    /// Break ties deterministically by ordering on burnchain block hash.
+    // PoX TODO: this method will need to be provided with a fork identifier
+    pub fn get_canonical_burn_chain_tip_stubbed(conn: &Connection) -> Result<BlockSnapshot, db_error> {
```

Comment:
Should this method take a connection to the PoX DB as well?

Reply:
Ultimately, this method is going to be removed -- I should update the TODO comment. The authoritative source for the burnchain tip is the burnchain db, not the sortition db.